client/src/setupTests.js

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import {
+  render,
+  fireEvent,
+  waitForElement,
+  cleanup
+} from "@testing-library/react";
+
+// localStorage mock for tests
+const mockLocalStorage = () => {
+  let store = {};
+  return {
+    getItem: function(key) {
+      return store[key] || null;
+    },
+    removeItem: function(key) {
+      delete store[key];
+    },
+    setItem: function(key, value) {
+      store[key] = value.toString();
+    },
+    clear: function() {
+      store = {};
+    }
+  };
+};
+
+global.localStorage = mockLocalStorage();
+global.render = render;
+global.fireEvent = fireEvent;
+global.waitForElement = waitForElement;
```

Comment:
I don't understand these last 3 assignments to `global`. Unless there's a strong reason for it, I think these should not become global and, instead, in `search.test.js` it should explicitly import the things it needs from `@testing-library/react`.